src/main/java/org/mule/extension/api/introspection/ImmutableRuntimeExtensionModel.java

```diff
@@ -0,0 +1,102 @@
+ */
+public final class ImmutableRuntimeExtensionModel extends ImmutableExtensionModel implements RuntimeExtensionModel
+{
+
+    private final Optional<ExceptionEnricherFactory> exceptionEnricherFactory;
+
+
+    /**
+     * Creates a new instance with the given state
+     *
+     * @param name                     the extension's name. Cannot be blank
+     * @param description              the extension's description
+     * @param version                  the extension's version
+     * @param vendor                   the extension's vendor name
+     * @param configurationModels      a {@link List} with the extension's {@link ConfigurationModel configurationModels}
+     * @param operationModels          a {@link List} with the extension's {@link OperationModel operationModels}
+     * @param connectionProviders      a {@link List} with the extension's {@link ConnectionProviderModel connection provider models}
+     * @param sourceModels             a {@link List} with the extension's {@link SourceModel message source models}
+     * @param modelProperties          A {@link Map} of custom properties which extend this model
+     * @param exceptionEnricherFactory an Optional @{@link ExceptionEnricherFactory} that creates a concrete {@link ExceptionEnricher} instance
+     * @throws IllegalArgumentException if {@code configurations} or {@link ParameterModel} are {@code null} or contain instances with non unique names, or if {@code name} is blank
+     */
+    public ImmutableRuntimeExtensionModel(String name,
+                                          String description,
+                                          String version,
+                                          String vendor,
+                                          List<ConfigurationModel> configurationModels,
+                                          List<OperationModel> operationModels,
+                                          List<ConnectionProviderModel> connectionProviders,
+                                          List<SourceModel> sourceModels,
+                                          Map<String, Object> modelProperties,
+                                          Optional<ExceptionEnricherFactory> exceptionEnricherFactory)
+    {
+        super(name, description,version,vendor,configurationModels,operationModels,connectionProviders,sourceModels, modelProperties);
+        this.exceptionEnricherFactory = exceptionEnricherFactory;
+    }
+
+
+    @Override
+    public Optional<ExceptionEnricherFactory> getExceptionEnricherFactory()
+    {
+        return exceptionEnricherFactory;
+    }
+
+    @Override
+    public String toString()
+    {
+        return new StringBuilder().append("ImmutableRuntimeExtensionModel{")
+                .append(super.toString())
+                .append(", exceptionEnricherFactory=").append(exceptionEnricherFactory)
+                .append('}').toString();
+    }
+
+    private <T extends Described> Map<String, T> toMap(List<T> objects)
```

Comment:
Do this belong here? They are private but are not called from within this class.